src/lib.rs

```diff
@@ -263,25 +265,27 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     /// TODO: as a high-level function, we will remove the fs_rng.
     fn batch_check<'a, R: RngCore>(
         vk: &Self::VerifierKey,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>> + Clone,
         query_set: &QuerySet<F>,
         evaluations: &Evaluations<F>,
         proof: &Self::BatchProof,
         rng: &mut R,
-        fs_rng: &mut Self::RandomOracle,
     ) -> Result<bool, Self::Error>
         where
             Self::Commitment: 'a,
     {
-        // TODO: as in batch_open(), setup Fiat-Shamir rng, etc.
+        let mut fs_rng = Self::RandomOracle::from_seed(&to_bytes![
+            query_set.clone().into_iter().map(|(_, (_, v))| v).collect::<Vec<_>>(),
+            commitments.clone().into_iter().collect::<Vec<_>>()
+        ].unwrap());
```

Comment:
as for fn batch_open()

Reply:
Previous functions state is restored as Daniele requested bellow